src/combinatorics/simple_graph/basic.lean

```diff
@@ -332,4 +332,41 @@ finset.max' (univ.image (λ (v : V), G.degree v)) (nonempty.image univ_nonempty
 
 end finite
 
+section complement
+
+/-!
+## Complement of a simple graph
+
+This section contains definitions and lemmas concerning the complement of a simple graph.
+-/
+
+/--
+We define `complement G` to be the `simple_graph V` such that no two adjacent vertices in `G`
+are adjacent in the complement, and every nonadjacent pair of vertices is adjacent
+(still ensuring that vertices are not adjacent to themselves.)
+-/
+def complement (G : simple_graph V) : simple_graph V :=
```

Comment:
I think we should rename this to `compl`, to be in line with other complements in mathlib, now that I think about it more.

Also add `@[simps]`.  This automatically gives you a simp lemma called `compl_adj`.

Reply:
After the `simps`, you can remove the rewriting with `complement` in the proof below as well, and use `compl_adj` instead.

Reply:
One minor issue I have with `compl` is it might be confused with `complete` but that's probably not really a big deal